src/components/layout/playground/requests/filters.tsx

```diff
@@ -274,9 +277,27 @@ const RequestFilters: React.FC<RequestFiltersProps> = ({
         </span>
         <FilterBy
           filters={otherFilters}
-          onFiltersChange={(newFilters) =>
-            onChange({ sort: filters.sort, ...newFilters })
-          }
+          onFiltersChange={(newFilters) => {
+            onChange({ sort: filters.sort, ...newFilters });
+            delete router.query['isPaidRequest'];
+            delete router.query['category'];
+            void router.push(
```

Comment:
The url should not be manipulated here, but in a `setFilter` handler in `src/pages/playground/index.tsx`. This is to keep the writing and reading to and from the url inside the same component.